Added the p:encode step

ndw opened the pull request:
Fix #659

In order to make the p:cast-content-type step decode any c:data document, even one that might contain serialized XML, I’ve reorganized the “Casting from an XML media type” section. The case of a c:data document is handled first.

This is a small backwards incompatibility because it means that you cannot use p:cast-content type to change the content type of a c:data element from application/xml to, for example, application/foo+xml. But that seems like something very unlikely given that we control the semantics of c:data documents. (The alternative is to create a p:decode step.)

xml-project reviewed (commented):
I think we shouldnt reuse XC0203 here. It is already used in p:http-request for an error constructing a multipart boundary.

xml-project commented:
@ndw Sorry, my fault. The spec uses XC0213, but I read XC0203. The later is an error in p:http-request.

ndw commented:
I went looking again in case I missed something and I found XC0052 in the error list, but I don't see it in the spec. It is in the `steps/payloads/request_body.xml` source, though. Now I'm wondering if there's part of the spec missing. I'll dig a little deeper tomorrow.

ndw commented:
We already had a defined, but unused because it was part of `p:http-request` that we deleted, error code for unsupported encoding. I don't see any reason to make up new error codes if we don't have to, so I've changed `err:XC0213` to `err:XC0052`.

ndw commented:
See also: https://github.com/xproc/3.0-steps/pull/666
